src/e2e/global-teardown.ts

```diff
@@ -0,0 +1,132 @@
+
+    try {
+      writeFileSync("full_test_report.json", existingFull, "utf-8");
+      writeFileSync("flake_test_report.json", existingFlake, "utf-8");
+      writeFileSync("failed_test_report.json", existingFailed, "utf-8");
+    } catch (err) {
+      console.error("Error writing to json", err);
+    }
+  } else {
+    console.log("[e2e_log] - Not a cron json, json report will not be created");
+  }
+}
+
+async function getExistingAndNewReports() {
+  let existingFullReports: CustomTestResult[] = [];
+  let existingFlakeReports: CustomTestResult[] = [];
+  let existingFailedReports: CustomTestResult[] = [];
+  let newFullReportJson: CustomTestResult[] = [];
+  let newFlakeReportJson: CustomTestResult[] = [];
+  let newFailReportJson: CustomTestResult[] = [];
+
+  try {
+    // fetch existing reports from repo
+    existingFullReports = (await fetchExistingReport(
+      "https://raw.githubusercontent.com/mozilla/blurts-server/e2e-report/full_test_report.json",
+    )) as CustomTestResult[];
+    existingFlakeReports = (await fetchExistingReport(
+      "https://raw.githubusercontent.com/mozilla/blurts-server/e2e-report/flake_test_report.json",
+    )) as CustomTestResult[];
+    existingFailedReports = (await fetchExistingReport(
+      "https://raw.githubusercontent.com/mozilla/blurts-server/e2e-report/failed_test_report.json",
+    )) as CustomTestResult[];
+
+    // new test generated files
+    const newFullReport = readFileSync("./new_full_test_report.json", "utf-8");
+    const newFlakeReport = readFileSync(
+      "./new_flake_test_report.json",
+      "utf-8",
+    );
+    const newFailReport = readFileSync(
+      "./new_failed_test_report.json",
+      "utf-8",
+    );
+
+    // parse new test reports
+    newFullReportJson = JSON.parse(newFullReport ?? "[]");
+    newFlakeReportJson = JSON.parse(newFlakeReport ?? "[]");
+    newFailReportJson = JSON.parse(newFailReport ?? "[]");
+
+    return {
+      existingFullReports,
+      existingFlakeReports,
+      existingFailedReports,
+      newFullReportJson,
+      newFlakeReportJson,
+      newFailReportJson,
+    };
+  } catch (err) {
+    console.log("[e2e_log] - error reading report files", err);
+    return {
+      existingFullReports,
+      existingFlakeReports,
+      existingFailedReports,
+      newFullReportJson,
+      newFlakeReportJson,
+      newFailReportJson,
+    };
+  }
+}
+
+async function fetchExistingReport(url: string) {
+  return new Promise((resolve, reject) => {
+    https
+      .get(url, (response) => {
+        let data = "";
+
+        response.on("data", (chunk) => {
+          data += chunk;
+        });
+
+        response.on("end", () => {
+          try {
+            const parsedData = JSON.parse(data);
+            resolve(parsedData);
+          } catch (err) {
+            reject(err);
+          }
+        });
+      })
+      .on("error", (err) => {
+        reject(err);
+      });
+  });
```

Comment:
Might just be easier to use `fetch()` API. Something like:

```js
return fetch(url).then(res => res.json());
```

Reply:
i believe that would require a third party